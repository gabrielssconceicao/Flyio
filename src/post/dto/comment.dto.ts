import { ApiProperty } from '@nestjs/swagger';

class UserComment {
  @ApiProperty({ example: 'John Doe', description: 'User name' })
  name: string;

  @ApiProperty({ example: 'jdoe', description: 'User username' })
  username: string;

  @ApiProperty({
    example: 'https://example.com/profile.jpg',
    description: 'User profile image URL',
  })
  profileImg: string | null;
}

export class CommentDto {
  @ApiProperty({ example: 'd-43-5df-df4', description: 'Comment identifier' })
  id: string;

  @ApiProperty({ example: 'Hello world', description: 'Comment content' })
  content: string;

  @ApiProperty({ example: '2023-03-15 14:30:00', description: 'Comment date' })
  createdAt: Date;

  @ApiProperty({ type: UserComment, description: 'Comment creator' })
  user: UserComment;
}
