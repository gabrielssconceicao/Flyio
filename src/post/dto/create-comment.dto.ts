import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MaxLength, MinLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreateCommentDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Comment text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Content', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Content', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Content', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Content', DtoErrorType.MAX_LENGTH, 255),
  })
  content: string;
}
