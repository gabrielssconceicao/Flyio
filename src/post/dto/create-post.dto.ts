import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MinLength, MaxLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreatePostDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Post text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Text', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Text', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Text', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Text', DtoErrorType.MAX_LENGTH, 255),
  })
  text: string;
}
