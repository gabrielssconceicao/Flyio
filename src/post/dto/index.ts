import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
