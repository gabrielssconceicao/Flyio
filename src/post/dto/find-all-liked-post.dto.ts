import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllLikedPostsResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts liked by the user',
  })
  items: PostEntity[];
}
