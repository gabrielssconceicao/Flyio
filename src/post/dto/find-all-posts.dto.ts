import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllUsersResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts of all users',
  })
  items: PostEntity[];
}
