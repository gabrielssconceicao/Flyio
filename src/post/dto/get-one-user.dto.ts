import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { CommentDto } from './comment.dto';

export class GetOnePostDto extends PostEntity {
  @ApiProperty({
    isArray: true,
    description: 'List of comments',
  })
  comments: CommentDto[];
}
