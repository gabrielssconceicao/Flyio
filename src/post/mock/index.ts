import { postMock } from './post.mock';
import { generateCreatePostDtoMock } from './create-post.dto.mock';
export { postMock, generateCreatePostDtoMock };
