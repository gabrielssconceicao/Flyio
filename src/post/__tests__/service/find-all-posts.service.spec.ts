import { Test, TestingModule } from '@nestjs/testing';
import { PrismaService } from 'src/prisma/prisma.service';
import { generateTokenPayloadDtoMock } from 'src/auth/mocks';
import { CloudinaryService } from 'src/cloudinary/cloudinary.service';
import { cloudinaryServiceMock } from 'src/cloudinary/mocks';
import { TokenPayloadDto } from 'src/auth/dto';
import { PostService } from '../../post.service';
import { generateFindAllPostsDtoMock } from '../../mock';
import { permissionServiceMock } from 'src/permission/mock/permission.service.mock';
import { PermissionService } from 'src/permission/permission.service';
describe('PostService - (findAll)', () => {
  let service: PostService;
  let prismaService: PrismaService;
  let cloudinary: CloudinaryService;
  let tokenPayload: TokenPayloadDto;
  let findAllPosts: ReturnType<typeof generateFindAllPostsDtoMock>;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        PostService,
        {
          provide: PrismaService,
          useValue: {
            findAll: jest.fn(),
          },
        },
        {
          provide: CloudinaryService,
          useValue: cloudinaryServiceMock,
        },
        {
          provide: PermissionService,
          useValue: permissionServiceMock,
        },
      ],
    }).compile();

    service = module.get<PostService>(PostService);
    prismaService = module.get<PrismaService>(PrismaService);
    cloudinary = module.get<CloudinaryService>(CloudinaryService);

    findAllPosts = generateFindAllPostsDtoMock();
    tokenPayload = generateTokenPayloadDtoMock();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
    expect(prismaService).toBeDefined();
    expect(cloudinary).toBeDefined();
  });

  it('should find all posts', async () => {
    jest.spyOn(prismaService, 'findAll').mockResolvedValue({
      count: findAllPosts.count,
      items: [
        {
          ...findAllPosts.items[0],
          PostLikes: 0,
          _count: { comments: 0, PostLikes: 0 },
        },
      ],
    });

    const result = await service.findAll(
      { limit: 10, offset: 0 },
      tokenPayload,
    );
    expect(prismaService.findAll).toHaveBeenCalled();
    expect(result).toEqual(findAllPosts);
    expect(result).toMatchSnapshot();
  });
});
