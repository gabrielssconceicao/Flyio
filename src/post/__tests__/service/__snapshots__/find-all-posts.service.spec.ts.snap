// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostService - (findAll) should find all posts 1`] = `
{
  "count": 1,
  "items": [
    {
      "commentsCount": 0,
      "content": "Hello world",
      "createdAt": 2023-03-15T17:30:00.000Z,
      "id": "42-d-f-df4",
      "images": [],
      "liked": false,
      "likes": 0,
      "user": {
        "profileImg": "https://example.com/profile.jpg",
        "username": "jDoe453",
      },
    },
  ],
}
`;
